src/modules/workspaces/workspaces.controller.ts

```diff
@@ -152,11 +155,136 @@ export class WorkspacesController {
   }
 
-  @Get('/:workspaceId/teams/:teamId/members')
+  @Get('/teams/:teamId/members')
+  @ApiOperation({
+    summary: 'Gets team members',
+  })
+  @ApiOkResponse({
+    description: 'Members of the team',
+  })
+  @UseGuards(WorkspaceGuard)
+  @WorkspaceRequiredAccess(AccessContext.WORKSPACE, WorkspaceRole.MEMBER)
+  async getTeamMembers(@Param('teamId') teamId: WorkspaceTeamAttributes['id']) {
+    if (!teamId) {
+      throw new BadRequestException('Invalid team ID');
+    }
+    return this.workspaceUseCases.getTeamMembers(teamId);
+  }
+
+  @Patch('/teams/:teamId')
+  @ApiOperation({
+    summary: 'Edits team data',
+  })
+  @ApiBearerAuth()
+  @ApiOkResponse({
+    description: 'Team has been edited',
+  })
+  @UseGuards(WorkspaceGuard)
+  @WorkspaceRequiredAccess(AccessContext.TEAM, WorkspaceRole.MANAGER)
+  async editTeam(
+    @Param('teamId') teamId: WorkspaceTeamAttributes['id'],
+    @Body() editTeamBody: EditTeamDto,
+  ) {
+    if (!teamId || !isUUID(teamId)) {
+      throw new BadRequestException('Invalid team ID');
+    }
+
+    return this.workspaceUseCases.editTeamData(teamId, editTeamBody);
+  }
+
+  @Delete('/teams/:teamId')
+  @ApiOperation({
+    summary: 'Deletes a team in a workspace',
+  })
+  @ApiBearerAuth()
+  @ApiOkResponse({
+    description: 'Deleted team',
+  })
+  @UseGuards(WorkspaceGuard)
+  @WorkspaceRequiredAccess(AccessContext.WORKSPACE, WorkspaceRole.OWNER)
+  async deleteTeam(@Param('teamId') teamId: WorkspaceTeamAttributes['id']) {
+    if (!teamId || !isUUID(teamId)) {
+      throw new BadRequestException('Invalid team ID');
+    }
+    return this.workspaceUseCases.deleteTeam(teamId);
+  }
+
+  @Post('/teams/:teamId/user/:userUuid')
+  @ApiOperation({
+    summary: 'Add a user to a team',
+  })
+  @ApiBearerAuth()
+  @ApiParam({ name: 'teamId', type: String, required: true })
+  @ApiParam({ name: 'userUuid', type: String, required: true })
+  @ApiOkResponse({
+    description: 'User added to team',
+  })
   @UseGuards(WorkspaceGuard)
-  @WorkspaceRequiredAccess(AccessContext.TEAM, WorkspaceRole.MEMBER)
-  async getTeamMembers() {
-    throw new NotImplementedException();
+  @WorkspaceRequiredAccess(AccessContext.TEAM, WorkspaceRole.MANAGER)
+  async addUserToTeam(
+    @Param('teamId') teamId: WorkspaceTeamAttributes['id'],
+    @Param('userUuid') memberId: UserAttributes['uuid'],
+  ) {
+    if (!teamId || !isUUID(teamId)) {
+      throw new BadRequestException('Invalid team id');
+    }
+
+    if (!memberId || !isUUID(memberId)) {
+      throw new BadRequestException('Invalid User Uuid');
+    }
+    const newTeamMember = await this.workspaceUseCases.addMemberToTeam(
+      teamId,
+      memberId,
+    );
+
+    return newTeamMember;
+  }
+
+  @Delete('/teams/:teamId/user/:userUuid')
+  @ApiOperation({
+    summary: 'Remove user from team',
+  })
+  @ApiBearerAuth()
+  @ApiParam({ name: 'teamId', type: String, required: true })
```

Comment:
The required: true does remove the necessity of the `if (!teamId)`, isn't it?

Reply:
It is for openAPI specifications, it does not affect the logic in the controller